operator/v1/types_ingress.go

```diff
@@ -68,6 +68,17 @@ type IngressControllerSpec struct {
 	//
 	// If empty, defaults to ingress.config.openshift.io/cluster .spec.domain.
 	//
+	// The domain value must be a valid DNS name. It must consist of lowercase
+	// alphanumeric characters, '-' or '.', and each label must start and end
+	// with an alphanumeric character and not exceed 63 characters.
+	//
+	// The total length of the domain is limited to 239 characters to account for
+	// the router name (e.g. "router-default.") that gets prepended when constructing
+	// the canonical hostname of which the total maximum length is 253 characters.
```

Comment:
I would put it this way:  
```suggestion
	// The total length of the domain name is limited to 253 characters,  inclusive of the 
	// prepended prefix starting with "router-".
```